spring-ws-core/src/main/java/org/springframework/ws/server/endpoint/observation/DefaultWebServiceEndpointConvention.java

```diff
@@ -0,0 +1,103 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.ws.server.endpoint.observation;
+
+import io.micrometer.common.KeyValue;
+import io.micrometer.common.KeyValues;
+import io.micrometer.observation.Observation;
+
+/**
+ * Default ObservationConvention for a WebService Endpoint.
+ * @author Johan Kindgren
+ */
+public class DefaultWebServiceEndpointConvention implements WebServiceEndpointConvention {
+
+    private static final KeyValue EXCEPTION_NONE = KeyValue.of(EndpointObservationDocumentation.LowCardinalityKeyNames.EXCEPTION, KeyValue.NONE_VALUE);
+    private String name = "webservice.server";
+
+
+    @Override
+    public KeyValues getLowCardinalityKeyValues(WebServiceEndpointContext context) {
+        return KeyValues.of(
+                exception(context),
+                localname(context),
+                namespace(context),
+                outcome(context),
+                soapAction(context));
+
+    }
+
+    private KeyValue localname(WebServiceEndpointContext context) {
+        return EndpointObservationDocumentation
+                .LowCardinalityKeyNames
+                .LOCALNAME
+                .withValue(context.getLocalname());
+    }
+
+    private KeyValue namespace(WebServiceEndpointContext context) {
+        return EndpointObservationDocumentation
+                .LowCardinalityKeyNames
+                .NAMESPACE
+                .withValue(context.getNamespace());
+    }
+
+
+    private KeyValue outcome(WebServiceEndpointContext context) {
+        return EndpointObservationDocumentation
+                .LowCardinalityKeyNames
+                .OUTCOME
+                .withValue(context.getOutcome());
+    }
+
+    private KeyValue soapAction(WebServiceEndpointContext context) {
+        return EndpointObservationDocumentation
+                .LowCardinalityKeyNames
+                .SOAPACTION
+                .withValue(context.getSoapAction());
+    }
+
+    private KeyValue exception(WebServiceEndpointContext context) {
+        if (context.getError() != null) {
+            return EndpointObservationDocumentation
+                    .LowCardinalityKeyNames
+                    .EXCEPTION
+                    .withValue(context.getError().getClass().getSimpleName());
+        } else {
+            return EXCEPTION_NONE;
+        }
+    }
+
+    @Override
+    public KeyValues getHighCardinalityKeyValues(WebServiceEndpointContext context) {
+        return KeyValues.empty();
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String getContextualName(WebServiceEndpointContext context) {
+        return "WebServiceEndpoint " + context.getNamespace() + ':' + context.getLocalname();
```

Comment:
Same remark as with the previous one

Reply:
I'm not sure how this value is used? What would be a good implementation in your opinion? This implementation describes the context as a WebServiceEndpoint that handles a payload with the given namespace and local-part.

Reply:
After reading https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#http-client-server-example, would it be ok to use the same ContextualName for a WebService Endpoint?
`POST /path`